Fixed issue #389

Manarnz opened the pull request:
Before, when the begin and end are AND/OR, switching back to Evolving Relationship would make the end "null." I fixed this issue.

Manarnz committed:
issue #375 And/Or end selections fixed

For issue #375, when the begin is and/or, the program used to make the end "no relationship." Now it is fixed, and users can select from template A.

Manarnz committed:
issue #375 And/Or end selections fixed

For issue #375, when the begin is and/or, the program used to make the end "no relationship." Now it is fixed, and users can select from template A.

Manarnz committed:
Merge branch 'Issue375' of https://github.com/amgrubb/BloomingLeaf into Issue375

Manarnz committed:
Update LinkInspector.js

Manarnz committed:
Issue #374 fixed in LinkInspector

Now, when begin is and/or the end will have template A, before that it will automatically assign end as "NO"

Manarnz committed:
Update LinkInspector.js

Manarnz committed:
LinkInspector.js issue #374 fixed

I fixed the bug that does not allow users to choose an end relationship when they select and/or as the begin.

Manarnz committed:
Responded to comments

Manarnz committed:
Issue #389 Fixed

Before, when the begin and end are AND/OR, switching back to Evolving Relationship would make the end "null." I fixed this issue.

leshghi commented on leaf-ui/rappid-extensions/LinkInspector.js:
```diff
@@ -228,14 +227,10 @@ var LinkInspector = Backbone.View.extend({
             $("#link-type-end").css("background-color","");
             
             //Saving this option
-            this.cell.prop("link-type", begin + "|" + end);
```
Why did you delete this line? Asking just because I don't know as much about link-type properties as you do, and just want to make sure we are still saving the type of link into the cell somewhere

Manarnz replied:
When I commented it out, it didn't make a difference because we have another line after it that does the same thing, so I deleted it because it is repetitive.

leshghi replied:
I looked but couldn't find the line you are referring to that does the same thing - could you direct me to it (just send me the number of the line in your branch)? Thanks!

Manarnz replied:
line 261, in updateEndEvolRelations. It is also important to note that the command I deleted was only in the AND/OR if statement, the other if statements did not include it, even the ones that have a begin and an end.

leshghi replied:
I see that, but as far as I can tell updateEndEvolRelations is not always called after updateBeginEvolRelations. For example, if someone changes the beginEvolRelationship, now that will not update the link-type property. Whether or not we need it to I think is a question we can work through as linkInspector.js is implemented with backbone, but in the meantime I think deleting this line is out of scope of the pull request so I would add it back in temporarily until we can look into it more

leshghi reviewed (commented):
Looks good to me - just answer my question below and once I understand why that line was deleted I'll approve so you can merge

leshghi reviewed (commented):
Also - did you mean to make this pull request for a merge into Issue374?

Manarnz commented:
> Also - did you mean to make this pull request for a merge into Issue374?

Yes, because it depends on the changes I made in issue374. is that okay?

leshghi commented:
> > Also - did you mean to make this pull request for a merge into Issue374?
> 
> Yes, because it depends on the changes I made in issue374. is that okay?

I don't think it should be dependent on your Issue374 branch, what part specifically is it dependent on?

Manarnz committed:
Added this.cell.prop("link-type", begin + "|" + end); back

Manarnz commented:
> > > Also - did you mean to make this pull request for a merge into Issue374?
> > 
> > 
> > Yes, because it depends on the changes I made in issue374. is that okay?
> 
> I don't think it should be dependent on your Issue374 branch, what part specifically is it dependent on?

Okay done!

Manarnz committed:
Fixed some overlapping changes

Because I created this branch off of another issue, the changes I made for that issue are in my code, so I just deleted them.

leshghi reviewed (changes requested):
When you click back into the linkInspector.js - although the text on the graph stays correct, the "end" dropdown menu gets reset to just say "end." We want it to say the same value as the link, so once you do that fix you should be set.

Manarnz committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into MAissue389

Manarnz committed:
Fixed the end placeholder bug

I deleted lines 239-246 because they saved end as "null" which created the problem. Line 239 showed end as null in the canvas when the user only selects a begin and lines 245-46 show SAVED when the user only selects a begin. I thought I cleared this error in my last debug but apparently I didn't.

Manarnz committed:
Revert "Fixed the end placeholder bug"

This reverts commit f1e907e9ee70b2822731110e43508112b7db9aa8.

Manarnz committed:
Fixed the end placeholder bug

I deleted lines 239-246 because they saved end as "null" which created the problem. Line 239 showed end as null in the canvas when the user only selects a begin and lines 245-46 show SAVED when the user only selects a begin. I thought I cleared this error in my last debug but apparently I didn't.

leshghi reviewed (approved):
Looks great! Congrats on finishing this one up :)
